work_queue/src/work_queue_worker.c

```diff
@@ -380,6 +383,14 @@ static void send_resource_update(struct link *manager)
 		total_resources->disk.total    = MAX(0, local_resources->disk.total);
 		total_resources->disk.largest  = MAX(0, local_resources->disk.largest);
 		total_resources->disk.smallest = MAX(0, local_resources->disk.smallest);
+		
+		//if workers are set to expire in some time, send the expiration time to manager
+		if(manual_wall_time_option > 0) {
+			end_time = worker_start_time + manual_wall_time_option;
+		}
+		else {
```

Comment:
No need for the else case here. (This eliminates confusion between the value set here and the default in line 117).